ner.py

```diff
@@ -75,13 +80,30 @@ def main():
     entity_types = set(entity_types)
     print("Extracting entities of types:", entity_types, "from '*.txt' documents in", args.data)
 
-    for path in glob.iglob(os.path.join(args.data, '*.txt')):
+    file_paths = glob.glob(os.path.join(args.data, '*.txt'))
+    num_files = len(file_paths)
+    print('Num files at path {}: {}'.format(args.data, num_files))
+    entities_for_file = {}
+    for idx, path in enumerate(file_paths):
         with open(path, 'r') as fhandle:
-            print("Processing:", fhandle.name)
-            doc = fhandle.read()
-            ents = extract_entities(doc, entity_types)
-            print("{}: {}".format(fhandle.name, ents))
+            try:
+                print("Processing:", fhandle.name)
+                doc = fhandle.read()
+                ents = extract_entities(doc, entity_types)
+                print("{}: {}".format(fhandle.name, ents))
+                entities_for_file[fhandle.name] = ents
+            except Exception as e:
```

Comment:
Catching everything could make it hard to trace down coding mistakes in the future. What about IOError instead?

Reply:
Processed 50k papers, only 8 were too big for spacy I assume. 
This was so we wouldn't process those 8 and the whole program wouldn't crash. Was a ValueError though so I'll change it to that